pycoast/cw_base.py

```diff
@@ -45,16 +45,11 @@ def get_resolution_from_area(area_def):
     """Get the best resolution for an area definition."""
     x_size = area_def.width
     y_size = area_def.height
-    prj = Proj(area_def.crs if hasattr(area_def, "crs") else area_def.proj_str)
-    if prj.is_latlong():
-        x_ll, y_ll = prj(area_def.area_extent[0], area_def.area_extent[1])
-        x_ur, y_ur = prj(area_def.area_extent[2], area_def.area_extent[3])
-        x_resolution = (x_ur - x_ll) / x_size
-        y_resolution = (y_ur - y_ll) / y_size
-    else:
-        x_resolution = (area_def.area_extent[2] - area_def.area_extent[0]) / x_size
-        y_resolution = (area_def.area_extent[3] - area_def.area_extent[1]) / y_size
+    x_resolution = abs(area_def.area_extent[2] - area_def.area_extent[0]) / x_size
+    y_resolution = abs(area_def.area_extent[3] - area_def.area_extent[1]) / y_size
     res = min(x_resolution, y_resolution)
+    if "degree" in area_def.crs.axis_info[0].unit_name:
+        res = _estimate_metered_resolution_from_degrees(area_def.crs, res)
```

Comment:
Ok, my bad, indeed, the `min(x_resolution, y_resolution)` saves the day, and the previous version was obviously wrong as it was returning a resolution in degrees. So this is probably good enough for our purposes.

However, in some cases, the current implementation could still be quite wrong I think (eg when the area is a stretched out horizontal rectangle just close to the pole). Maybe the use of `pyproj.Geod`to compute distances would improve the accuracy in general? But no need to do it in this PR.